src/main/java/com/couchbase/lite/replicator/PusherInternal.java

```diff
@@ -55,6 +55,8 @@
 
     public static final int MAX_PENDING_DOCS = 200;
 
+    private static final int TIMEOUT_FOR_PAUSE = 30 * 1000; // 30 sec
```

Comment:
Not sure 30 seconds is too long or not. Bringing to down to ~5 seconds might be a good idea (shouldn't cost performance impact).